src/reference/asciidoc/ftp.adoc

```diff
@@ -620,6 +622,89 @@ public class FtpJavaApplication {
 
 Notice that, in this example, the message handler downstream of the transformer has an advice that removes the remote file after processing.
 
+[[ftp-rotating-server-advice]]
+=== Inbound Channel Adapters: Polling Multiple Servers and Directories
+
+Starting with _version 5.0.7_, the `RotatingServerAdvice` is available; when configured as a poller advice, the inbound adapters can poll multiple servers and directories.
+Configure the advice and add it to the poller's advice chain as normal.
+A `DelegatingSessionFactory` is used to select the server see <<ftp-dsf>> for more information.
+The advice configuration consists of a list of `RotatingServerAdvice.KeyDirectory` objects.
+
+.Example
+[source, java]
+----
+@Bean
+public RotatingServerAdvice advice() {
+    List<KeyDirectory> keyDirectories = new ArrayList<>();
+    keyDirectories.add(new KeyDirectory("one", "foo"));
+    keyDirectories.add(new KeyDirectory("one", "bar"));
```

Comment:
Oh! I see why `Map.Entry` is not good for us: the `key` can be the same.
So, well maybe your original proposition was fine. Together with the natural `equals()` and `hashCode()` to rely on.